src/librustc_driver/driver.rs

```diff
@@ -1353,7 +1353,8 @@ fn generated_output_paths(
     exact_name: bool,
     crate_name: &str,
 ) -> Vec<PathBuf> {
-    let mut out_filenames = Vec::new();
+    let mut out_filenames = Vec::with_capacity(sess.opts.output_types.len());
```

Comment:
This doesn't appear to be correct.

Reply:
It's later shrunk at 1380; processing `keys()` before the loop (which was in my initial commit) caused a really uninformative error to occur during stage 1 build.

Reply:
Then it's a better idea not to touch it at all. This is in the driver, and runs once per compilation, AFAICT, so it's not really worth optimizing (unless you can rewrite it to an iterator, that is).